include/lsst/meas/modelfit/RegularizedMoments.h

```diff
@@ -0,0 +1,123 @@
+ *  maximally likely image moments given moments measured from an image. As such the class
+ *  provides two methods in addition to the at method already mentioned. One method computes the
+ *  biased weighted moments given the specified weight and "true" moments, and the other computes
+ *  the derivative of the objective function along each of the image moment variables.
+ */
+class MomentsModel {
+public:
+    using Moments = Eigen::Matrix<double, 6, 1>;
+    using Jacobian = Eigen::Matrix<double, 6, 6>;
+    using FirstMoment = Eigen::Matrix<double, 2, 1>;
+    using SecondMoment = Eigen::Matrix<double, 2, 2>;
+
+    /**
+     * Construct a MomentsModel object with a given vector of weight function moments
+     *
+     * @param[in]   W       An eigen vector of the moments of the elliptical Gaussian weighting function.
+     *                      Most likely the first element (the zeroth moment) should be one, representing
+     *                      a normalized weight function.
+     */
+    MomentsModel(Moments const & W): W(W) {
+    }
+
+    /**
+     *  Sets the moments corresponding to the astronomical object's intrinsic moments
+     *
+     *  When this function is called, it sets the astronomical object's intrinsic moments for all subsequent
+     *  calculations. Additionally after setting the moment, this function calculates and caches the value
+     *  of the biased weighted moments given the weights and intrinsic moments.
+     *
+     *  @param[in]  Q       An eigen vector of the intrinsic moments of the astronomical object
+     */
+    void at(Moments const & Q);
+
+    /**
+     *  Returns the value of the biased weighted moments given the input weight and object's intrinsic moments
+     *
+     *  @param[out]         Eigen vector of the model biased weighted moments given the input weight and
+                            intrinsic moments
+     */
+    Moments computeValues();
+
+    /**
+     *  Computes and returns the gradient of the model biased weighted moments along each of the intrinsic
+     *  moment dimensions.
+     *
+     *  @param[out]         A six by six Eigen matrix, with the biased weighted moments down the column, and
+     *                      the derivative of the column moment with respect to the intrinsic moments along
+     *                      the rows.
+     */
+    Jacobian computeJacobian();
+
+private:
+    // Calculates the Model evaluated at the input intrinsic moments given the specified weighted moments
+    void makeValue();
+
+    // Storage variables
+    Moments W, Q, value;
+
+    double constant;
+
+    FirstMoment alpha;
+
+    SecondMoment beta;
+};
+
+// Tests for classes in anonymous name spaces
+bool testConstant(double tol=1e-6);
+
+bool testAlphaX(double tol=1e-6);
+bool testAlphaY(double tol=1e-6);
+
+bool testBetaX(double tol=1e-6);
+bool testBetaY(double tol=1e-6);
+bool testBetaXY(double tol=1e-6);
```

Comment:
I don't know what our convention is, but if these are actually for testing they should be declared elsewhere (in the tests).

Reply:
Also you may want to `double constexpr DEFAULT_TEST_TOLERANCE = 1.e-6` somewhere.

Reply:
I added in the constexpr, and discussed with you and Jim about the best place for this type of test, and it was decided that these will remain here until some possible point in the future when a policy is put in place.